MoreCyclopsUpgrades/CyclopsUpgrades/UpgradeHandler.cs

```diff
@@ -0,0 +1,139 @@
+﻿namespace MoreCyclopsUpgrades.CyclopsUpgrades
+{
+    using Common;
+    using System;
+    using System.Collections.Generic;
+
+    public delegate UpgradeHandler UpgradeHandlerCreateEvent();
+    public delegate void UpgradeEvent(SubRoot cyclops);
+    public delegate void UpgradeEventSlotBound(SubRoot cyclops, Equipment modules, string slot);
+
+    /// <summary>
+    /// Represents all the behaviors for a cyclops upgrade module at the time of the module being installed and counted.
+    /// </summary>
+    public class UpgradeHandler
+    {
+        /// <summary>
+        /// The TechType that identifies this type of upgrade module.
+        /// </summary>
+        public readonly TechType techType;
+
+        private int count = 0;
```

Comment:
The `count` checking done in this class doesn't work for TieredUpgradeHandlers.  
Instead of having to override all the methods, perhaps instead this class could have an internal field that informs if `count` should be ignored or not.